src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py

```diff
@@ -247,6 +248,23 @@ def __eq__(self, other: "SnowflakePlan") -> bool:
     def __hash__(self) -> int:
         return hash(self._id) if self._id else super().__hash__()
 
+    @property
```

Comment:
nit: do you consider `cache_property` for slight perf improvement? what if every call needs a re-calculation then it doesn't apply

Reply:
oh, that can not be a cache_property here, because optimizations will be applied based on configurations, and the queries generated can be different based on the optimization configuration at the time when this property is called. 
For example:
session.cte_optimizationz_enabled = False
queries1 = df._plan.execution_queries
session.cte_optimizationz_enabled = True
queries2= df._plan.execution_queries

queries1 and queries2 are expected to be different here. If we use cached_property, queries1 and queries2 will be the same, which will not be correct.

Reply:
I see, thanks for the explanation!